tenant/store.go

```diff
@@ -76,84 +79,64 @@ func (tenantStore *tenantStore) listSegments(tenantId uint64) ([]Segment, error)
 }
 
 func (tenantStore *tenantStore) addTenant(tenant *Tenant) error {
+	log.Println("In tenantStore addTenant().")
+
 	var tenants []Tenant
 	tenantStore.DbStore.Db.Find(&tenants)
 
-	//	myId := tenant.Id
+	// Most UUID generators return , or "" if uuid is
+	// invalid thus len != 32 should be rejected.
+	// Create new UUID in case one is not provided.
+	if len(tenant.UUID) != 32 {
```

Comment:
Why do we need to allow the creation of tenants that cannot be mapped to the external system? In both OpenStack and Kubernetes we have external tenant identifiers that should and could be used.